src/lib.rs

```diff
@@ -476,22 +500,69 @@ fn optimize_png(
 
     // Do this first so that reductions can ignore certain chunks such as bKGD
     perform_strip(png, opts);
-    let stripped_png = png.clone();
```

Comment:
Why remove this variable? Its value was used in the `if let Some((opts, idat_data)) = best { ... }` statement at what's now the `optimize_raw` function, so maybe this introduced an unintended behavior change?

Reply:
`stripped_png` was used to revert back to the original state if reductions failed. Because of the way the functions have been refactored, it's no longer necessary to revert anything - the value of `png` here in `optimize_png` will only be modified if reductions were successful.